internal/api_server/server.go

```diff
@@ -99,8 +106,27 @@ func (s *Server) Run(ctx context.Context) error {
 
 	router := chi.NewRouter()
 
+	requestSizeLimiter := func() func(http.Handler) http.Handler {
+		return func(next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if len(r.URL.String()) > maxUrlLength {
+					http.Error(w, fmt.Sprintf("URL too long, exceeds %d characters", maxUrlLength), http.StatusRequestURITooLong)
+					return
+				}
+				if len(r.Header) > maxNumHeaders {
+					http.Error(w, fmt.Sprintf("Request has too many headers, exceeds %d", maxNumHeaders), http.StatusRequestEntityTooLarge)
+					return
+				}
+
+				next.ServeHTTP(w, r)
+			})
+		}
+	}
+
 	// general middleware stack for all route groups
 	router.Use(
+		middleware.RequestSize(1024*1024), // Limit request body size to 1MB
```

Comment:
That seems a bit tight. In our validation code, we allow inline configs and certificates that are larger than this.
https://github.com/flightctl/flightctl/blob/main/api/v1alpha1/validation.go#L16-L17

Reply:
Good point.  Certificates can be up to 20MB?
What do you suggest for this value?  50MB?